[Spike] Add `evm` contents (#12)
futureversecom/trn-docs: docs/evm/getting-started.md

$XRP is a 6 dp token, while Ethereum tooling like MetaMask and Solidity expect native token amounts in wei or 18dp units. To keep compatibility simple, TRN takes care of scaling inputs down by `1e-12` and scaling up outputs by `1e12`. Any fractional amounts < `1e12` / `0.000001 $XRP` are always rounded up only by adding `1e12`

$ROOT token is available as en ERC20 token at `0xcCcCCccC00000001000000000000000000000000`. $ROOT is also a 6dp token but does not require any special handling as per the ERC20 standard.

:::tip

The `@therootnetwork/evm` package exports a handy method `assetIdToERC20Address` that converts the native asset id to EVM contract address, in $ROOT case, it's effectively `assetIdToERC20Address(1)` with $ROOT asset id is `1`

:::

#### Transfer Examples

1. Transfers `1 $XRP` to `receiver`

```solidity
receiver.call{value: 1 ether}("")
```

2. Transfers `1.00001 $XRP` to `receiver` (i.e rounds up by `0.00001 $XRP`)

```solidity
receiver.call{value: 1 ether + 1 wei}("")
```

> Similarly, the following will transfers `0.00001 $XRP` to `receiver`

Reviewer: ```suggestion
Similarly, the following will transfer `0.00001 $XRP` to `receiver`
```

```typescript
await wallet.sendTransaction({
  to: receiver,
  value: 1,
});
```

> or transfers `1.2 $XRP` to `receiver`

Reviewer: ```suggestion
or transfer `1.2 $XRP` to `receiver`
```

```typescript
await wallet.sendTransaction({
  to: receiver,
  value: ethers.utils.parseEther("1.2"),
});
```

:::note Recap

- $XRP is the default "native" currency
- Input and output amounts for $XRP on EVM are expressed in `wei` (18dp) as normal
- Amounts transferred `< 1e12` are always rounded up

:::

### Transaction Fees

TRN supports EIP-1559 and legacy transaction fee models. Ethereum tooling is able to query fee estimation info accurately out of the box without any special handling required. The [`eth_feeHistory`](https://docs.alchemy.com/alchemy/apis/ethereum/eth_feehistory) RPC is provided for this purpose see its docs for more details.

### Addresses

EVM contract developers on TRN can use existing Ethereum wallets and addresses without any changes. The network support ECDSA address by default.

### Runtime Pallets from EVM (Precompiles)

> TRN uses a precompile system to allow communication between the TRN runtime and EVM contracts. This allows native TRN currencies and NFTs to appear as regular ERC20s and ERC721s to Ethereum tooling and contracts along side with other functionality to supercharge contract development such as as Decentralised Exchange (DEX), FuturePass, ERC1155.

Reviewer 2: ```suggestion
TRN exposes functionalities of certain pallets as precompiles, which allows users/tooling to query/transact with the pallet functions as if they are solidity functions. This also allows native TRN currencies and NFTs to appear as regular ERC20s and ERC721s to Ethereum tooling and contracts along side with other functionality to supercharge contract development such as as Decentralised Exchange (DEX), FuturePass, ERC1155.
```
